base/tps/src/main/java/org/dogtagpki/server/tps/rest/TPSCertService.java

```diff
@@ -162,6 +161,52 @@ public Response findCerts(String filter, String tokenID, Integer start, Integer
 
             return createOKResponse(response);
 
+        } catch (Exception e) {
+            throw new PKIException(e.getMessage());
+        }
+    }
+
+    private Response findAllCerts(String filter, Integer start, Integer size) {
+        String method = "TPSCertService:findAllCerts: ";
+        String msg = "";
+        TPSCertCollection response = new TPSCertCollection();
+
+        logger.debug("TPSCertService.findAllCerts({}, {}, {})", filter, start, size);
+
+        TPSEngine engine = TPSEngine.getInstance();
+        try {
+            TPSSubsystem subsystem = (TPSSubsystem) engine.getSubsystem(TPSSubsystem.ID);
+            TPSCertDatabase certDatabase = subsystem.getCertDatabase();
+            Iterator<TPSCertRecord> certRecs = certDatabase.findRecords(filter).iterator();
+            TokenDatabase tokenDatabase = subsystem.getTokenDatabase();
+            List<String> authorizedProfiles = getAuthorizedProfiles();
+            if (authorizedProfiles == null) {
+                msg = "authorizedProfiles null";
+                logger.debug("{}{}", method, msg);
+                throw new PKIException(method + msg);
+            }
+
+            int i = 0;
+            // skip to the start of the page
+            for (; i < start && certRecs.hasNext(); i++) certRecs.next();
+
+            if (certRecs.hasNext()) {
+                certRecs.forEachRemaining(certRecord -> {
+                    int j = 0;
```

Comment:
The `j` should indicate the index of each entry in the current page, so it should not be a constant.

Paging works like this:
```
all entries:  0 --------|================|----------------------- <total>
current page:         <start>     <start + size>
````

Reply:
Thanks for the explanation. In light of the pagination, I dropped the `forEachRemianing` implementation in favour of an implementation that mimics the original more closely.